query/src/main/java/org/infinispan/query/backend/QueryKnownClasses.java

```diff
@@ -151,40 +154,55 @@ void stop() {
       }
    }
 
-   boolean containsKey(final Class<?> clazz) {
+   boolean containsKey(Class<?> clazz) {
       if (indexedEntities != null) {
          return indexedEntities.contains(clazz);
       }
       return localCache.get().containsKey(clazz);
    }
 
-   Boolean get(final Class<?> clazz) {
+   Boolean get(Class<?> clazz) {
       if (indexedEntities != null) {
          return indexedEntities.contains(clazz);
       }
       return localCache.get().get(clazz);
    }
 
-   void put(final Class<?> clazz, final Boolean value) {
+   void put(Class<?> clazz, Boolean value) {
       if (indexedEntities != null) {
-         throw new IllegalStateException("Autodetect mode is not enabled");
+         throw new IllegalStateException("Autodetect mode is not enabled for cache " + cacheName);
       }
 
       if (value == null) {
          throw new IllegalArgumentException("Null values are not allowed");
       }
+
       startInternalCache();
+
       Transaction tx = transactionHelper.suspendTxIfExists();
       try {
-         runCommand(() -> knownClassesCache.put(new KeyValuePair<>(cacheName, clazz), value));
+         // put and retry if SuspectException was thrown
+         while (true) {
+            try {
+               knownClassesCache.put(new KeyValuePair<>(cacheName, clazz), value);
+               break;
+            } catch (CacheException e) {
+               if (SuspectException.isSuspectExceptionInChain(e)) {
```

Comment:
This shouldn't be needed, when we get a `SuspectException` we retry internally (in `StateTransferInterceptor`)